src/MotionControl.c

```diff
@@ -1208,6 +1208,12 @@ BOOL Ros_MotionControl_StartMotionMode(MOTION_MODE mode)
         }
     }
 
+    if (Ros_Controller_IsAnyFaultActive())
+    {
+        Ros_Debug_BroadcastMsg("Controller is in a fault state. Please call /reset_error");
+        return FALSE;
+    }
+
```

Comment:
I would perhaps suggest to move this up, to right after the call to `Ros_Controller_IsRemote()`.

Doesn't seem much point to 'wait this long' to check whether there is an active alarm if we could check early on and just bail out.

It should still work in case we're just trying to get servos out of eco mode, as that would not work with an active alarm/error in any case.

Reply:
Updated